genesis-tools/genesis-csv-to-json/src/csv_parser.rs

```diff
@@ -0,0 +1,347 @@
+//! Constructs state of token holders from the csv file.
+use chrono::{DateTime, Utc};
+use csv::ReaderBuilder;
+use near::config::AccountInfo;
+use near_crypto::{KeyType, PublicKey};
+use near_network::PeerInfo;
+use near_primitives::hash::{hash, CryptoHash};
+use near_primitives::receipt::{ActionReceipt, Receipt, ReceiptEnum};
+use near_primitives::serialize::to_base64;
+use near_primitives::transaction::{Action, FunctionCallAction};
+use near_primitives::types::{AccountId, Balance, Gas};
+use near_primitives::utils::create_nonce_with_nonce;
+use near_primitives::views::{AccessKeyPermissionView, AccessKeyView, AccountView};
+use node_runtime::StateRecord;
+use serde::{Deserialize, Serialize};
+use std::fs::File;
+use std::io::Read;
+use std::path::PathBuf;
+
+/// Methods that can be called by a non-privileged access key.
+const REGULAR_METHOD_NAMES: &[&str] = &["stake", "transfer"];
+/// Methods that can be called by a privileged access key.
+const PRIVILEGED_METHOD_NAMES: &[&str] = &["add_access_key", "remove_access_key"];
+/// Methods that can be called by an access key owned by a "foundation".
+const FOUNDATION_METHOD_NAMES: &[&str] =
+    &["add_access_key", "remove_access_key", "permanently_unstake", "terminate", "init"];
+/// Amount of gas that we pass to run the contract initialization function.
+const INIT_GAS: Gas = 1_000_000;
+
+type Result<T> = std::result::Result<T, Box<dyn std::error::Error>>;
```

Comment:
same, let's keep the actual std::io::Error here

Reply:
We can keep `std::io::Error` for `verify()` function but other functions in this file mix errors, e.g. `std::io::Error` with https://docs.rs/csv/1.1.1/csv/struct.Error.html . Since this is a standalone tool that it well-encapsulated I am less concerned of boxed error leaking into the rest of the infrastructure.